### DISCUSSION TITLE
Have Printing time information not only when printing from SD card, but also when printing via serial / USB

### DISCUSSION THREAD

**Discussion by heinerolli** [mriscoc/Ender3V2S1#892] [Ideas]
Hi Miguel; 
first of all thank you for this great firmware! 
It works flawless for me. (I am using Ender3V2-422-BLTUBL-MPC-20230312.bin)
Recently, I added the ESP3D project to my printer, which allows 'remote printing' (via an ESP32 wired to the serial port within the Ender) Also this works great, 
However, when I print like that via serial, I do not get printing time information on the Enders screen (which I really miss).
It turns out that that behavior is ruled by the following setting in Configuration_adv.h (https://github.com/mriscoc/Ender3V2S1/blob/Ender3V2S1-Released/Marlin/Configuration_adv.h):

```
  //#define M73_REPORT                    // Report M73 values to host
  #if BOTH(M73_REPORT, SDSUPPORT)
    #define M73_REPORT_SD_ONLY            // Report only when printing from SD
  #endif
```

My understing is that if you set:
`    #define M73_REPORT_SD_ONLY            // Report only when printing from SD`
to 
`    #define M73_REPORT            // Report only when printing from SD`

-> Is that something you could consider implementing in your precompiled versions?
I would be happy to test that, but I am not able to do the compilation on my own...
I am relying on your precomplied versions

Hmmm - just saw that things like that should be possible via patreon . Will look ito this.

(*) There are workarounds like injecting M73 information during slicing, but this is only a band-aid

**Comment by mriscoc** [comment]
Include the **`M75 {File name}`** at the start of your g-code file.

Check https://github.com/mriscoc/Ender3V2S1/wiki/Octoprint for useful scripts for host printing.

**Comment by heinerolli** [comment]
Thanks, Miguel!
I‘ll try that.
One question: At the time when  I put

M75 {File name}

in my printer start gcode, I do not know the actual file name which is to be printed. Is there some variable / placeholder for

File name

which I could use in the above instruction?

All the best, Heiner


Am 29.04.2023 um 18:57 schrieb Miguel Risco-Castillo ***@***.***>:

﻿

Include the M75 {File name} at the start of your g-code file.

Check https://github.com/mriscoc/Ender3V2S1/wiki/Octoprint for useful scripts for host printing.

**Reply by mriscoc** [reply]
{File name} totally depends on your host system, each one has its own way. If you can't find how to get the filename from your host, just leave it as: `M75 Printing from host`
